src/topology/category/Top/limits.lean

```diff
@@ -125,6 +126,452 @@ instance forget_preserves_colimits : preserves_colimits (forget : Top.{u} ⥤ Ty
+
+@[simp, reassoc]
+lemma pi_iso_pi_inv_π {ι : Type u} (α : ι → Top) (i : ι) :
+  (pi_iso_pi α).inv ≫ pi.π α i = ⟨λ f, f i, continuous_apply i⟩ :=
+by simpa [pi_iso_pi]
+
+@[simp]
+lemma pi_iso_pi_inv_π_apply {ι : Type u} (α : ι → Top.{u}) (i : ι) (x : Π i, α i) :
+  (pi.π α i : _) ((pi_iso_pi α).inv x) = x i :=
+concrete_category.congr_hom (pi_iso_pi_inv_π α i) x
+
+@[simp]
+lemma pi_iso_pi_hom_apply {ι : Type u} (α : ι → Top.{u}) (i : ι) (x : ∏ α) :
+  (pi_iso_pi α).hom x i = (pi.π α i : _) x :=
+begin
+  have := pi_iso_pi_inv_π α i,
+  rw iso.inv_comp_eq at this,
+  exact concrete_category.congr_hom this x
+end
+
+/-- The explicit cofan of a family of topological spaces given by the sigma type. -/
+@[simps X ι_app]
+def sigma_cofan {ι : Type u} (α : ι → Top.{u}) : cofan α :=
+@cofan.mk _ _ _ α (Top.of (Σ i, α i)) (λ b, ⟨sigma.mk b⟩)
+
+/-- The constructed cofan is indeed a colimit -/
+def sigma_cofan_is_colimit {ι : Type u} (α : ι → Top.{u}) : is_colimit (sigma_cofan α) :=
+{ desc := λ S,
+  { to_fun := λ s, S.ι.app s.1 s.2,
+    continuous_to_fun := by { apply continuous_sigma, intro i, dsimp, continuity } },
+  fac' := by tidy,
+  uniq' := begin
+    intros S m h,
+    ext ⟨i, x⟩,
+    simp [← h i]
+  end }
+
+/--
+The coproduct is homeomorphic to the disjoint union of the topological spaces.
+-/
+def sigma_iso_sigma {ι : Type u} (α : ι → Top.{u}) : ∐ α ≅ Top.of (Σ i, α i) :=
+(colimit.is_colimit _).cocone_point_unique_up_to_iso (sigma_cofan_is_colimit α)
+
+@[simp, reassoc]
+lemma sigma_iso_sigma_hom_ι {ι : Type u} (α : ι → Top) (i : ι) :
+  sigma.ι α i ≫ (sigma_iso_sigma α).hom = ⟨sigma.mk i⟩ :=
+by simp [sigma_iso_sigma]
+
+@[simp]
+lemma sigma_iso_sigma_hom_ι_apply {ι : Type u} (α : ι → Top) (i : ι) (x : α i) :
+  (sigma_iso_sigma α).hom ((sigma.ι α i : _) x) = sigma.mk i x :=
+concrete_category.congr_hom (sigma_iso_sigma_hom_ι α i) x
+
+@[simp]
+lemma sigma_iso_sigma_inv_apply {ι : Type u} (α : ι → Top) (i : ι) (x : α i) :
+  (sigma_iso_sigma α).inv ⟨i, x⟩ = (sigma.ι α i : _) x :=
+by { rw [← sigma_iso_sigma_hom_ι_apply, ← comp_app], simp, }
+
+
+lemma limit_induced (F : J ⥤ Top.{u}) :
+  (limit F).topological_space = ⨅ j, (F.obj j).topological_space.induced (limit.π F j) :=
+begin
+  let homeo := homeo_of_iso (limit.iso_limit_cone ⟨_, limit_cone_is_limit F⟩),
+  refine homeo.inducing.induced.trans _,
+  dsimp only [is_open, limit_cone, of, Top.topological_space, subtype.topological_space,
+    Pi.topological_space],
+  simp only [induced_infi, induced_compose],
+  congr
+end
+
+section prod
+
+lemma prod_topology {X Y : Top} :
+  (X ⨯ Y).topological_space =
+    induced (limits.prod.fst : X ⨯ Y ⟶ _) X.topological_space ⊓
+      induced (limits.prod.snd : X ⨯ Y ⟶ _) Y.topological_space :=
+begin
+  conv_lhs { rw Top.limit_induced },
+  apply le_antisymm,
+  rw le_inf_iff,
+  exact ⟨infi_le _ walking_pair.left, infi_le _ walking_pair.right⟩,
+  rw le_infi_iff,
+  rintro (_|_),
+  exacts [inf_le_left, inf_le_right]
+end
```

Comment:
If you move this to somewhere after `prod_iso_prod`, you could use that isomorphism to obtain this easily.